### ISSUE TITLE
Inverse for .throw method

### ISSUE THREAD

**Issue by Janpot** [petkaantonov/bluebird#483]
I found `.throw(reason)` is a [shorthand](https://github.com/petkaantonov/bluebir`d/blob/master/API.md#throwdynamic-reason---promise) for

```js
.then(function() {
   throw reason;
});
```

I was looking for something like `.ignoreError(value)` (needs a better name) which would be shorthand for:

```js
.catch(function() {
   return value;
});
```

So with this you could basically 'invert' the logic

```js
somePromise
  .ignoreError('use this value instead')
  .throw(new Error('the original promise was expected to error'))
```

Does this pattern already exist? I might be looking over it in the docs

**Comment by Janpot** [comment]
or even `.ignoreError(Type, value)` for

```js
.catch(Type, function() {
   return value;
});
```

**Comment by benjamingr** [comment]
To be fair - this is something I've had to do a lot, and it'd also be nice for people who don't want possibly unhandled rejections.

I wonder if we can come up with a shorter more concise name.

**Comment by Janpot** [comment]
@benjamingr Yes please find a better name :)

**Comment by spion** [comment]
There is `.thenReturn`, `.thenThrow`, I propose `.catchReturn` and `.catchThrow` for completeness :)

**Comment by benjamingr** [comment]
@spion but the aliases are so useful as in `.return` and in `.throw`

**Comment by Janpot** [comment]
I just realized this wouldn't work, the `.throw(` would pick up the resolved promise from `.ignoreError(`
```js
somePromise
  .ignoreError('use this value instead')
  .throw(new Error('the original promise was expected to error'))
```
If this inversion is a use case maybe there can be an `.invert` method too?

**Comment by benjamingr** [comment]
@petkaantonov  name?

**Comment by petkaantonov** [comment]
probably gonna be catchThrow and catchReturn, with same signature as catch (accepts predicates)

**Comment by tjconcept** [comment]
Would this ease something like:
```js
.catch(predicate, function( e ){
  doLogging();
  throw e;
});
```
Or would that require another method, something like `catchTap`?

**Comment by Janpot** [comment]
@tjconcept not going to happen: https://github.com/petkaantonov/bluebird/issues/517

**Comment by tjconcept** [comment]
@Janpot well, logging was just an easy example, in my most recent case I'm using it to augment all types of a specific error with more context.

**Comment by petkaantonov** [comment]
@tjconcept augmenting seems like a valid use case actually

**Comment by benjamingr** [comment]
@petkaantonov so is logging (although augmenting is more common for sure) but I wonder if it's worth the extra sugar this requires in the library.

**Comment by petkaantonov** [comment]
Fixed in 3.0 release
